File: go/bob.go
package bob

import (
	"regexp"
	"strings"
)

func Hey(remark string) string {
	re := regexp.MustCompile(`\s`)
	remark = re.ReplaceAllString(remark, "")
	if remark == "" {
		return "Fine. Be that way!"
	}
	isQuestion := strings.HasSuffix(remark, "?")
	re = regexp.MustCompile(`[A-Z]`)
	isYelling := re.MatchString(remark) && strings.ToUpper(remark) == remark
	if isQuestion && isYelling {
		return "Calm down, I know what I'm doing!"
	}

	if isQuestion {
		return "Sure."
	}

	if isYelling {
		return "Whoa, chill out!"
	}

	return "Whatever."
}

File: ocaml/bob.ml
open Base

let response_for s =
  let s = String.strip s in
  let letters = String.filter s ~f:Char.is_alpha in
  let is_question = String.is_suffix s ~suffix:"?" in
  let is_yelling =
    (not (String.is_empty letters))
    && String.equal (String.uppercase letters) letters
  in
  if String.is_empty s then "Fine. Be that way!"
  else if is_question && is_yelling then "Calm down, I know what I'm doing!"
  else if is_question then "Sure."
  else if is_yelling then "Whoa, chill out!"
  else "Whatever."
